sklearn/inspection/_partial_dependence.py

```diff
@@ -25,10 +31,12 @@
 from sklearn.ensemble._hist_gradient_boosting.gradient_boosting import (
     BaseHistGradientBoosting)
 
-
```

Comment:
We need to add a docstring

```
def_get_predictions(...):
    """Get predictions for an estimator by replacing the values of
    a specific feature, keeping others constant.

    Parameters
    ----------
    estimator : estimator object
        The scikit-learn predictor.
    grid : ndarray of shape (n_points, n_target_features)
        The different values to try for the targeted features.
    features_indices : ndarray of shape (n_targeted_features,)
        The indices of the features targeted. The values of these
        features will be changed.
    X : {ndarray, dataframe} of shape (n_samples, n_features)
        The dataset to be used.
    response_method : {"auto", "predict_proba", "decision_function"}
        The response method to use by the estimator. It applies only for classifier.
    """
```

We should rename est to estimator

Reply:
The shape of the array need to be checked to be sure.

Reply:
Done in https://github.com/scikit-learn/scikit-learn/pull/16164/commits/58630130d1639431591c7b75773466099ab75983.

Parameter validation is already happening at https://github.com/scikit-learn/scikit-learn/blob/58630130d1639431591c7b75773466099ab75983/sklearn/inspection/_partial_dependence.py#L229-L230
and 
https://github.com/scikit-learn/scikit-learn/blob/58630130d1639431591c7b75773466099ab75983/sklearn/inspection/_partial_dependence.py#L281-L288
Do you want any additional validation to be performed?